code/src/componenets/ThoughtForm.js

```diff
@@ -0,0 +1,35 @@
+import React from 'react'
+
+const ThoughtForm = ({onFormSubmit, newThought, setNewThought, setCount, count}) => {
+
+const checkKey = (event) => {
+    if (event.keyCode === 13 && !event.shiftKey) {
+        onFormSubmit(event)
+    }
+}
+
+    return (
+        <div className="form-container">
+        <form onSubmit={onFormSubmit}>
+          <label htmlFor="newThought">What's making you happy right now?</label>
+          <textarea 
+            id="newThought" 
+            type="text"
+            value={newThought} 
+            onChange={(e) => setNewThought(e.target.value) || setCount(e.target.value.length)}
```

Comment:
I was wondering how you could add two different functions, and now i see that i didn't have to create a third function combining two